From aef2534fb1bfef9ac3befd49b59abb94bcd3246d Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Thu, 30 Sep 2021 10:20:56 +0900
Subject: [PATCH 3/4] starterkit: linux: rom-writer: add speed_up support

As same as windows version, we can speed up rom-writer
using 'sup' command.
But, file transfer with 'send' command is bottleneck.
This patch change file transfer to use file redirection,
and supports speed_up.

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 starterkit/linux/rom_writer | 40 +++++++++++++++++++++++++++----------
 1 file changed, 29 insertions(+), 11 deletions(-)

diff --git a/starterkit/linux/rom_writer b/starterkit/linux/rom_writer
index 29b9327..50d8989 100755
--- a/starterkit/linux/rom_writer
+++ b/starterkit/linux/rom_writer
@@ -84,20 +84,12 @@ proc sw_change_back {} {
 }
 
 proc upload_file {file} {
-    #
-    # read srec file
-    #
-    set f [open $file]
-    set srec [split [read $f] "\n"]
-    close $f
-
+    global tty
     #
     # send srec as ascii
     #
-    set send_slow {10 .001}
-    foreach s $srec {
-	send -s "$s\n"
-    }
+    exec cat $file > $tty
+    send "\n"
 }
 
 proc file_check {lists} {
@@ -169,6 +161,31 @@ proc cpld_mode {} {
     }
 }
 
+proc speed_up {} {
+    global tty
+    set baudrate "115200"
+
+    send "\r"
+    expect {
+	">" { send "sup\r" }
+    }
+    expect {
+	"460.8Kbps" { set baudrate "460800" }
+	"921.6Kbps" { set baudrate "921600" }
+    }
+
+    # close current minicom and re-open minicom
+    close; wait
+    spawn env LANG=C minicom -D $tty -b $baudrate
+
+    expect {
+	"Press CTRL-A Z for help on special keys" { send "\r" }
+    }
+    expect {
+	">"
+    }
+}
+
 proc before_minicom {} {
     global mode
 
@@ -186,6 +203,7 @@ proc after_minicom {} {
     } else {
 	cpld_mode
     }
+    speed_up
 }
 
 proc run_main {lists} {
-- 
2.31.1

